File: readability.py
import cs50

# get input
text = cs50.get_string('Text: ')

# variables
words = x = l = s = 0

# loop entire string
for x in range(len(text)):
    # count letters
    if text[x].isalnum():
        l += 1
    # count words
    if text[x].isspace():
        words += 1
    # count sentences
    if text[x] in ['.', '?', '!']:
        s += 1

words += 1
L = l * 100 / words
S = s * 100 / words
grade = 0.0588 * L - 0.296 * S - 15.8

if grade >= 16:
    print('Grade 16+')
    
elif grade < 1:
    print('Before Grade 1')
        
else:
    print(f'Grade {round(grade)}')

File: readability.c
//Libraries
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

//Variables
string text;
int letters, sentences, spaces, i, grades, words;
void letcount();
float index, L, S;


//Main
int main()
{
    //Get Text
    text = get_string("Text: ");
    //Letter Counter
    letcount();
    
    words = (spaces + 1);
    
    //Grade assignments
    L = ((float)letters / (float)words * 100);
    
    S = ((float)sentences / (float)words * 100);
    
    index = (0.0588 * L - 0.296 * S - 15.8);
    
    grades = round(index);
    
    //printf("Letters: %i\nWords: %i\nSentences: %i\n", letters, words, sentences);
    
    //Grade
    if (grades < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grades > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grades);
    }

}

//Count Characters
void letcount()
{
    for (i = 0; i < strlen(text); i++)
    {
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            letters++;
        }
        if (text[i] == ' ')
        {
            spaces++;
        }
        if (text[i] == '?' || text[i] == '.' || text[i] == '!')
        {
            sentences++;
        }
    }
}